reconcile/database_access_manager.py

```diff
@@ -0,0 +1,620 @@
+                                    },
+                                },
+                                {
+                                    "name": "db.user",
+                                    "valueFrom": {
+                                        "secretKeyRef": {
+                                            "name": job_data.rds_admin_secret_name,
+                                            "key": "db.user",
+                                        },
+                                    },
+                                },
+                            ],
+                            "resources": {
+                                "requests": {
+                                    "cpu": "100m",
+                                    "memory": "128Mi",
+                                },
+                            },
+                            "volumeMounts": [
+                                {
+                                    "name": "configs",
+                                    "mountPath": "/tmp/scripts/",
+                                    "readOnly": True,
+                                },
+                            ],
+                        },
+                    ],
+                    "volumes": [
+                        {
+                            "name": "configs",
+                            "projected": {
+                                "sources": [
+                                    {
+                                        "secret": {
+                                            "name": job_data.script_secret_name,
+                                        },
+                                    },
+                                ],
+                            },
+                        },
+                    ],
+                },
+            },
+        },
+    }
+    return OpenshiftResource(
+        body=job,
+        integration=QONTRACT_INTEGRATION,
+        integration_version=QONTRACT_INTEGRATION_VERSION,
+    )
+
+
+def get_service_account_spec(name: str) -> OpenshiftResource:
+    return OpenshiftResource(
+        body={
+            "apiVersion": "v1",
+            "kind": "ServiceAccount",
+            "metadata": {
+                "name": name,
+                "labels": {
+                    "app": "qontract-reconcile",
+                    "integration": QONTRACT_INTEGRATION,
+                },
+            },
+            "automountServiceAccountToken": False,
+        },
+        integration=QONTRACT_INTEGRATION,
+        integration_version=QONTRACT_INTEGRATION_VERSION,
+    )
+
+
+class DBAMResource(BaseModel):
+    resource: OpenshiftResource
+    clean_up: bool
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class JobStatusCondition(BaseModel):
+    type: str
+
+
+class JobStatus(BaseModel):
+    conditions: list[JobStatusCondition]
+
+    def is_complete(self) -> bool:
+        return True if self.conditions else False
```

Comment:
the purpose of this function is to figure out if the job is not running anymore (at least i understand that from the usage), regardless out outcome right?

a successful job also has a condition, namely `type=Complete`

Reply:
nevermind. i checked and it seems that's the case. a running job has no conditions and only gets one when succeeding or failing.

Reply:
there are some job types where there might be conditions in between (index runs) but we are not using them here so should be ok

Reply:
I think that does not apply here, since this only checks if the job is still running. Not caring about the exit status